File: SerpentineApi/SerpentineApi/Responses/ChannelResponse.cs
namespace SerpentineApi.Responses;

public class ChannelResponse : BaseResponse
{

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public bool AdultContent { get; set; } = false;

    public ChannelMemberResponse MyMember { get; set; } = new();

    public int MembersCount { get; set; } = 0;

    public string CoverPicture { get; set; } = "";

    public string BannerPicture { get; set; } = "";

    public int UnreadMessages { get; set; } = 0;


   
    
    



}

File: serpentineclient/src/models/responses/channel-response.ts
import { BaseResponse } from "./base-response";
import { ChannelMemberResponse } from "./channel-member-response";

export class ChannelResponse extends BaseResponse {
  name: string = '';
  description: string = '';
  adultContent: boolean = false;
  myMember: ChannelMemberResponse | null = null;
  membersCount: number = 0;
  coverPicture: string | null = null;
  bannerPicture: string | null = null;
  unreadMessages: number = 0;
  readonly userIsOwner: boolean = false;


  //front-end-only properties
  isReaded: boolean = false; //this property is set when the message bubble gets into the viewport for the first time

  
 constructor(init?: Partial<ChannelResponse>) {
    super(init); 
    Object.assign(this, init);
    this.userIsOwner = this.myMember?.isOwner ?? false;
  }


  
}